gril/grilunsol.c

```diff
@@ -519,11 +519,16 @@ int g_ril_unsol_parse_signal_strength(GRil *gril, const struct ril_msg *message,
 	int lte_signal;
 	int signal;
 
+	gboolean samsung_quirks = (g_ril_vendor(gril) == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x);
+
 	g_ril_init_parcel(message, &rilp);
 
 	/* RIL_SignalStrength_v5 */
 	/* GW_SignalStrength */
 	gw_sigstr = parcel_r_int32(&rilp);
+	/* Samsung on MSM822x does need a weird masking here. From vendor's java file. */
```

Comment:
"Vendor's java file" implies that you have access to Samsung's source code (which you don't). Refers to LOS's code instead (and preferably the link).

```suggestion
	/*
	 * Samsung on MSM822x needs weird masking here, according to LOS's RIL
	 * for this device.
	 * https://github.com/LineageOS/android_device_samsung_msm8226-common/blob/cm-12.1/ril/SamsungMSM8226RIL.java
	 */
```